src/main/java/uk/gov/companieshouse/company/links/processor/InsolvencyStreamProcessor.java

```diff
@@ -116,11 +131,22 @@ public void processDelta(Message<ResourceChangedData> resourceChangedMessage) {
         logger.trace(String.format("Performing a PATCH with "
                 + "company number %s for contextId %s", companyNumber, logContext));
 
-        if (links == null) {
-            links = new Links();
+        if (links == null || links.getInsolvency() == null) {
```

Comment:
getInsolvency can be blank string, probably better verify using StringUtils.isBlank

Reply:
Relevant code change done